### DISCUSSION TITLE
In celery update is commit but not working

### DISCUSSION THREAD

**Discussion by ZhaoGuan** [sqlalchemy/sqlalchemy#10119] [Usage Questions]
### Describe the bug

In celery update is commit but not working
![image](https://github.com/sqlalchemy/sqlalchemy/assets/29194004/d8d1c947-d59e-462b-b6c7-bb732e0b0506)

### SQLAlchemy Version in Use

2.0.19

### DBAPI (i.e. the database driver)

pymysql

### Database Vendor and Major Version

mysql 5.7

### Python Version

3.9.11

### Operating system

OSX

### To Reproduce

```python
class PlanResultSql(BaseSql):
    def __init__(self, db_session):
        super().__init__(PlanTaskResult, the_session=db_session)


    def update_plan_result(self, tag, is_finish, result):
        print("UPDATE!")
        print(self.select(select_action=select(self.model)).all())
        update_action = update(self.model).where(self.model.tag == tag).values(
            {"result": result, "is_finish": is_finish})
        try:
            self.session.execute(update_action)
            self.session.flush()
            self.session.commit()
            return True
        except:
            traceback.print_exc()
            self.session.rollback()
            return False
```

### Error

```
# Copy the complete stack trace and error message here, including SQL log output if applicable.
```

**Comment by zzzeek** [comment]
nothing seems wrong with the code and the screenshot of logging shows COMMIT, which is when SQLAlchemy emits `.commit()` on the DBAPI connection.

**Reply by ZhaoGuan** [reply]
Yes but it is not work.

**Reply by CaselIT** [reply]
Without a reproduction example it's hard to suggest anything
